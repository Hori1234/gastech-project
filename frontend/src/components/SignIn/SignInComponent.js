import React, { useState } from "react";
import { useHistory, useLocation } from 'react-router-dom'
import { useAuth } from "../contextConfig";
import {
  Layout,
  message,
  Avatar,
  Form,
  Input,
  Typography,
  Divider,
  Button,
  Checkbox,
} from "antd";
import "antd/dist/antd.css";
import { UserOutlined } from "@ant-design/icons";

const { Title } = Typography;

export default function SignInComponent() {
  const auth = useAuth();
  let history = useHistory();
  let location = useLocation();
  let { from } = location.state || { from: { pathname: "/" } };

  const [state, setState] = useState({
    username: "",
    password: "",
    remember: false
  });

  const onAuthenticate = async () => {
    const success = await auth.login({
      username: state.username,
      password: state.password,
      remember: state.remember
    });

    if (success) {
      history.replace(from);
    }

    else {
      message.info("Account not valid");
    }
  };

  const handleChangeUsername = (event) => {
    setState({
      ...state,
      username: event.target.value
    });
  };

  const handleChangePassword = (event) => {
    setState({
      ...state,
      password: event.target.value
    });
  };

  const handleChangeRemember = (event) => {
    setState({
      ...state,
      remember: event.target.value
    })
  }

  const layout = {
    labelCol: {
      span: 8,
    },
    wrapperCol: {
      span: 16,
    },
  };

  const tailLayout = {
    wrapperCol: {
      offset: 8,
      span: 16,
    },
  };

  return (
    <Layout
      style={{
        margin: 30,
        height: "95vh",
        display: "flex",
        backgroundColor: "white",
      }}
    >
      <Layout
        style={{
          alignItems: "center",
          marginTop: 15,
          backgroundColor: "white",
        }}
      >
        <Avatar
          style={{ marginTop: 100 }}
          shape="square"
          size={64}
          icon={<UserOutlined />}
        />
        <Divider style={{ marginTop: 10 }}>Welcome</Divider>
      </Layout>

      <Layout
        style={{
          alignItems: "center",
          display: "flex",
          marginTop: -70,
          marginBottom: -120,
          backgroundColor: "white",
        }}
      >
        <Title style={{ alignItems: "center" }} level={5.2}>
          Sign In
        </Title>
        <Title style={{ alignItems: "center" }} level={5}>
          Fill the form to log in
        </Title>
      </Layout>

      <Layout
        style={{
          alignItems: "center",
          display: "flex",
          marginTop: -50,
          backgroundColor: "white",
        }}
      >
        <Form {...layout} name="basic" initialValues={{ remember: true }}>
          <Form.Item
            label="Username"
            name="username"
            rules={[{ required: true, message: "Please input your username!" }]}
            style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
          >
            <Input value={state.username} onChange={handleChangeUsername} />
          </Form.Item>

          <Form.Item
            label="Password"
            name="password"
            rules={[{ required: true, message: "Please input your password!" }]}
            style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
          >
            <Input.Password
              value={state.password}
              onChange={handleChangePassword}
            />
          </Form.Item>

          <Form.Item
            {...tailLayout}
            name="remember"
            style={{ width: "100%", marginRight: 180, marginLeft: -50 }}
          >
            <Checkbox
              checked={state.remember}
              onChange={handleChangeRemember}
            >Remember me
            </Checkbox>
          </Form.Item>

          <Form.Item {...tailLayout}>
            <Button
              type="primary"
              htmlType="submit"
              style={{ marginLeft: 10, marginTop: 60, width: "60%" }}
              onClick={onAuthenticate}
            >
              Login
            </Button>
          </Form.Item>
        </Form>
      </Layout>
    </Layout>
  );
}
