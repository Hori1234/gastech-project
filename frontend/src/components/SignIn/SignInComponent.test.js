import React from "react";
import SignInComponent from "./SignInComponent";
import {shallow} from "enzyme";

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn().mockReturnValue({
      pathname: '/login',
      search: '',
      hash: '',
      state: null,
      key: '5nvxpbdafa',
    }),
    useHistory: jest.fn().mockReturnValue({
      push: jest.fn()
    })
}));

describe("SignInComponent", () => {
  const component = shallow(<SignInComponent/>);

  it("shall render the login page correctly", () => {
    expect(component).toMatchSnapshot();
  });

});
