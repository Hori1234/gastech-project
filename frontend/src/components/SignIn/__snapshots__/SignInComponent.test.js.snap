// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInComponent shall render the login page correctly 1`] = `
<Layout
  style={
    Object {
      "backgroundColor": "white",
      "display": "flex",
      "height": "95vh",
      "margin": 30,
    }
  }
>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "marginTop": 15,
      }
    }
  >
    <Avatar
      icon={<ForwardRef(UserOutlined) />}
      shape="square"
      size={64}
      style={
        Object {
          "marginTop": 100,
        }
      }
    />
    <Divider
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      Welcome
    </Divider>
  </Layout>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "marginBottom": -120,
        "marginTop": -70,
      }
    }
  >
    <Title
      level={5.2}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      Sign In
    </Title>
    <Title
      level={5}
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      Fill the form to log in
    </Title>
  </Layout>
  <Layout
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "display": "flex",
        "marginTop": -50,
      }
    }
  >
    <ForwardRef(InternalForm)
      initialValues={
        Object {
          "remember": true,
        }
      }
      labelCol={
        Object {
          "span": 8,
        }
      }
      name="basic"
      wrapperCol={
        Object {
          "span": 16,
        }
      }
    >
      <FormItem
        label="Username"
        name="username"
        rules={
          Array [
            Object {
              "message": "Please input your username!",
              "required": true,
            },
          ]
        }
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
      >
        <Input
          onChange={[Function]}
          type="text"
          value=""
        />
      </FormItem>
      <FormItem
        label="Password"
        name="password"
        rules={
          Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ]
        }
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
      >
        <Password
          action="click"
          iconRender={[Function]}
          onChange={[Function]}
          value=""
          visibilityToggle={true}
        />
      </FormItem>
      <FormItem
        name="remember"
        style={
          Object {
            "marginLeft": -50,
            "marginRight": 180,
            "width": "100%",
          }
        }
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Checkbox
          checked={false}
          indeterminate={false}
          onChange={[Function]}
        >
          Remember me
        </Checkbox>
      </FormItem>
      <FormItem
        wrapperCol={
          Object {
            "offset": 8,
            "span": 16,
          }
        }
      >
        <Button
          block={false}
          ghost={false}
          htmlType="submit"
          loading={false}
          onClick={[Function]}
          style={
            Object {
              "marginLeft": 10,
              "marginTop": 60,
              "width": "60%",
            }
          }
          type="primary"
        >
          Login
        </Button>
      </FormItem>
    </ForwardRef(InternalForm)>
  </Layout>
</Layout>
`;
